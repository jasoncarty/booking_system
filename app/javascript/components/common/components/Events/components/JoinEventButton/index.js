import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { sendRequest } from './../../../../utils';

class JoinEventButton extends Component {
  static propTypes = {
    onSuccess: PropTypes.func.isRequired,
    onError: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired
  }

  handleClick = () => {
    const {
      id,
      onSuccess,
      onError
    } = this.props;

    sendRequest({
      url: `/events/${id}/book`,
      type: 'post',
      onSuccess,
      onError
    });
  }

  render() {
    return (
      <button
        type="button"
        className="btn btn-blue btn-center"
        onClick={this.handleClick}
      >
        Join in
      </button>
    );
  }
}

export default JoinEventButton;
