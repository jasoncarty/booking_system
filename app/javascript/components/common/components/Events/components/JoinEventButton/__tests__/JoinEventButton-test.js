import React from 'react';
import { create } from 'react-test-renderer';
import JoinEventButton from './../index';

const mockSendRequest = jest.fn();
jest.mock('./../../../../../utils/sendRequest', () => () => mockSendRequest);

const defaultProps = {
  onSuccess: () => {},
  onError: () => {},
  id: 1
};

describe('<JoinEventButton/>', () => {
  it('matches snapshot with default props', () => {
    const element = create(
      <JoinEventButton {...defaultProps} />
    );
    expect(element).toMatchSnapshot();
  });

  it('calls sendRequest', () => {
    const element = create(
      <JoinEventButton {...defaultProps} />
    );
    const instance = element.getInstance();
    const {
      onSuccess,
      onError,
      id
    } = defaultProps;

    instance.handleClick();
    setTimeout(() => {
      expect(mockSendRequest).toBeCalledWith({
        url: `/events/${id}/book`,
        type: 'post',
        onSuccess,
        onError
      });
    }, 0);
  });
});
