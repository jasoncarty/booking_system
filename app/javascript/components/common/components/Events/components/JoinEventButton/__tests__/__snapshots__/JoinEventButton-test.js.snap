// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JoinEventButton/> matches snapshot with default props 1`] = `
<button
  className="btn btn-blue btn-center"
  onClick={[Function]}
  type="button"
>
  Join in
</button>
`;
